docs/oci-storage.md

```diff
@@ -1,40 +1,152 @@
+* Reduce etcd load by offloading large bundle content.  
+* Use a standardized storage backend for large manifests or Helm charts.
 
-This may be interesting for users who need to store big `Bundles`, and could also be seen as the first step for an `OCIOps` feature in the future.
+Once the OCI registry is enabled, Fleet will use it as the source for storing Bundle resources. When Fleet can't access the OCI registry, it won't fall back to default etcd storage. Instead, it logs errors.
 
-Once the OCI registry is enabled, Fleet will use it as the source for storing `Bundle` resources.
-When Fleet can't access the OCI registry, it won't fall back to default `etcd` storage. Instead, it will log errors so they can be fixed.
+![A visual asset displaying the flow of Fleet with OCI Storage.](../static/img/fleet-ociStorage-flow.png)
 
-## Configuring the OCI registry
+## **Prerequisites**
 
-OCI registry values should be configured as an extra section in the `GitRepo` yaml.
+* A running OCI registry.  
+* A Kubernetes secret with valid credentials.  
+* A Fleet installation (v2.12.0 or later) .
 
-There are the fields involved:
+## Creating an OCI Storage Secret
+
+To enable OCI Storage, you have to define a specific kind of secret. There are two ways of defining secrets
+
+* **Global secret (ocistorage)** : Applies to all GitRepos in the namespace.  
+  * This is the fallback secret. If no GitRepo-level secret is specified, Fleet uses this secret for all GitRepos in the namespace.  
+* **GitRepo-level secret (ociRegistrySecret):** – Overrides the global secret for a specific GitRepo.  
+  * This is a user-defined secret that you reference in the `GitRepo` resource. 
+
+:::note
+If the referenced secret doesn't exist or contains invalid credentials, Fleet logs an error and skips deployment. It doesn't fall back to etcd.
+:::
+
+### GitRepo Configuration Example
+
+The GitRepo spec is extended with information about the OCI Registry. You must update your GitRepo resource to include a reference to the secret:
+
+```yaml
+metadata:
+  name: coffee-shop-deployments
+  namespace: fleet-local
+spec:
+  repo: https://github.com/example/coffee-shop-fleet
+  branch: main
+  paths:
+    - ./manifests
+  ociRegistrySecret: ocistorage
+```
+
+The `ociRegistrySecret` references a secret in the same namespace, which contains the address and credentials needed to read/write to the OCI registry. 
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ocistorage
+  namespace: fleet-local
+type: fleet.cattle.io/bundle-oci-storage/v1alpha1
+data:
+  reference: <base64-encoded-registry-url>
+  username: <base64-encoded-user>
+  password: <base64-encoded-password>
+  insecureSkipTLS: <base64-encoded-true/false>
+  basicHTTP: <base64-encoded-true/false>
+# Optional
+  agentUsername: <base64-encoded-readonly-user>
+  agentPassword: <base64-encoded-password>
+```
+:::note
+If your user is not defined in the spec of the Repo.Then Fleet tries to use default credentials.
```

Comment:
What does this mean? 
Default credentials means no username/password? 

I would also mention that only the `reference` field is required, the rest are optional.
If no username/password are defined then no authentication will be used to access the registry.